### PR TITLE
feat(TDP-4508) : backport fix on maintenance cloud

### PR DESCRIPTION
**Pull request by tlnd-jbrun**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4508

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/955/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/955/)

File: dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java [hunk 1]
```diff
@@ -380,6 +388,39 @@ public String copy(String preparationId, String name, String destination) throws
     }
 
     /**
+     * Duplicate the list of steps and set it to the new preparation
+     *
+     * @param originalPrep the original preparation.
+     * @param targetPrep the created preparation.
+     */
+    private void cloneStepsListBetweenPreparations(Preparation originalPrep, Preparation targetPrep) {
```
Comment (hmdebenque) [inline dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java]: this duplicates copyStepsFrom(String id, String from) line 577

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/999/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/999/)

File: dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java (hunk 1 above)
Comment (tlnd-jbrun) [reply dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java]: the change asked is done

**Pull request merged**